### PR TITLE
auto doc pr

### PR DESCRIPTION
**Pull request by pengyuan-zhou**
## Description
Please include a summary of the change, the motivation, and any additionl context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: .github/workflows/docpr.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          rm -rf dokka
+          ./gradlew dokka
+          
+      - name: Commit files
+        run: |
+          git add -f docs
+          git diff-index --quiet HEAD || git commit -m "Auto Document PR"
+     
+      - name: Push changes
+        run: git push
```
Comment (pengyuan-zhou) [inline .github/workflows/docpr.yml:40]: maybe here should use `git push origin HEAD:<name-of-remote-branch>`
but Idk how to write the branch name here to map to the supposed one.

Comment (vkkhare) [reply .github/workflows/docpr.yml:40]: I think it would be 
`git push origin HEAD:dev`

File: .github/workflows/docpr.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: AutoDocumentPR
+
+on:
+  push:
+    branches: 
+      - dev
+      - master
```
Comment (vkkhare) [inline .github/workflows/docpr.yml:7]: run the action only on push to dev

File: .github/workflows/docpr.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          restore-keys: ${{ runner.os }}-gradle-
+
+      - name: Setup JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+          
+      - name: Run dokka
+        run: |
+          rm -rf dokka
```
Comment (vkkhare) [inline .github/workflows/docpr.yml:31]: what is this supposed to delete? There is no directory by the name dokka. Dokka is just a gradle action

**Comment by pengyuan-zhou** [comment]
> Try running this action on a test. One way of testing this would be to run this action on pull-request and commit to a test branch

Have tested on my folk before pushing the last commit.

**Comment by pengyuan-zhou** [comment]
> Try running this action on a test. One way of testing this would be to run this action on pull-request and commit to a test branch

Ok, now tested [here](https://github.com/FL-Wearable/KotlinSyft/actions/runs/188415458), seems fine.

**Comment by vkkhare** [comment]
> > Try running this action on a test. One way of testing this would be to run this action on pull-request and commit to a test branch
> 
> Ok, now tested [here](https://github.com/FL-Wearable/KotlinSyft/actions/runs/188415458), seems fine.

Doesn't seem to work. The git push origin dev will push to openmined/kotlinsyft dev which ideally you should not be able to do. Additionally it says already up to date. Most probably this would be happening because of no changes in doc files. Maybe just document a small file to see the doc changes are actually different

**Comment by pengyuan-zhou** [comment]
> > > Try running this action on a test. One way of testing this would be to run this action on pull-request and commit to a test branch
> > 
> > 
> > Ok, now tested [here](https://github.com/FL-Wearable/KotlinSyft/actions/runs/188415458), seems fine.
> 
> Doesn't seem to work. The git push origin dev will push to openmined/kotlinsyft dev which ideally you should not be able to do. Additionally it says already up to date. Most probably this would be happening because of no changes in doc files. Maybe just document a small file to see the doc changes are actually different

[Test](https://github.com/pengyuan-zhou/Kotlintest/blob/dev/docs/syftlib/org.openmined.syft/-syft/testautodoc.md)

**Comment by vkkhare** [comment]
closing for refactor branch rule. Will reopen

**Comment by vkkhare** [comment]
Hi this seems to work but I have doubts as to why did it push to your local dev instead of openmined/KotlinSyft dev?

**Comment by pengyuan-zhou** [comment]
> Hi this seems to work but I have doubts as to why did it push to your local dev instead of openmined/KotlinSyft dev?
> 
> I think the push command should be
> `git push -u origin HEAD`

Hi, if you are referring to the [Test](https://github.com/pengyuan-zhou/Kotlintest/blob/dev/docs/syftlib/org.openmined.syft/-syft/testautodoc.md), because that's a duplicate instead of a folk, so the origin there is that repo instead of this one.

**Comment by vkkhare** [comment]
Ahh okay then just rebase the pr then and I will merge

**Comment by pengyuan-zhou** [comment]
> Ahh okay then just rebase the pr then and I will merge

Hmmm, I could squash those on my folk, but here seems problematic, maybe due to other new commits during this PR. I will close this one and remake a clean PR.

**Pull request closed without merging**